src/term/mod.rs

```diff
@@ -1081,7 +1081,17 @@ pub enum UnaryOp {
     ///
     /// It's also worth noting that [`UnaryOp::DeepSeq`] should be, in principle, more efficient that [`UnaryOp::Force`]
     /// as it does less cloning.
-    Force(),
+    ///
+    /// # About `for_export`
+    ///
+    /// When exporting a Nickel term, we first apply `Force` to the term to
+    /// evaluate it. If there are record fields that have been marked `not_exported`,
+    /// they would still be evaluated ordinarily, see [#1230](https://github.com/tweag/nickel/issues/1230).
+    /// To stop this from happening, we introduce the `for_export` parameter
+    /// here. When `for_export` is `true`, the evaluation of `Force` will skip
+    /// fields that are marked as `not_exported`. When `for_export` is `false`,
+    /// these fields are evaluated.
+    Force { for_export: bool },
```

Comment:
Nitpick: although the documentation is nice and explains it well, I tend to think that the naming could reflect the _behavior_ of the two different variant for `Force`, instead of the motivation. That is, `for_export` doesn't tell much to a newcomer on the codebase (and the meaning could even evolve). On the other hand, something like `ignore_not_exported` is more specific, IMHO.